lib/elixir/lib/float.ex

```diff
@@ -106,11 +108,15 @@ defmodule Float do
       34.25
 
   """
-  @spec floor(float, 0..15) :: float
-  def floor(number, precision \\ 0) when is_float(number) and precision in 0..15 do
+  @spec floor(float, precision_range) :: float
+  def floor(number, precision \\ 0) when is_float(number) and precision in @precision_range do
     round(number, precision, :floor)
   end
 
+  def floor(number, precision) when is_float(number) do
+    raise ArgumentError, "precision #{inspect precision} is out of valid range of #{inspect @precision_range}"
```

Comment:
You can refactor the message to a private function, it is important only for the raise call to be on this function. :)